File: sdk/javascript/src/guidance.ts
export type GuidanceRequest = {
  agent_id: string
  domain: string
  intent: string
  context?: Record<string, unknown>
  risk_level?: 'low' | 'medium' | 'high'
}

export class Guidance {
  constructor(readonly guidance: Record<string, any>) {}

  toPrompt(): string {
    const seedSkills = this.guidance.seed_skills ?? []
    const candidateSkills = this.guidance.candidate_skills ?? []
    const verifiedSkills = this.guidance.verified_skills ?? []

    return [
      'Runtime Guidance',
      renderSection('Verified Skills', verifiedSkills),
      renderSection('Candidate Skills', candidateSkills),
      renderSection('Seed Skills', seedSkills),
    ].join('\n\n')
  }
}

export class GuidanceResource {
  constructor(private readonly baseUrl: string) {}

  async get(request: GuidanceRequest): Promise<Guidance> {
    const response = await fetch(`${this.baseUrl}/v1/guidance`, {
      method: 'POST',
      headers: { 'content-type': 'application/json' },
      body: JSON.stringify({
        context: {},
        risk_level: 'low',
        ...request,
      }),
    })

    if (!response.ok) {
      throw new Error(`Guidance request failed: ${response.status}`)
    }

    const body = await response.json()
    return new Guidance(body.guidance)
  }
}

function renderSection(title: string, skills: Array<Record<string, any>>): string {
  if (!skills.length) {
    return `${title}:\nNone.`
  }

  const rendered = skills.map((skill, index) => {
    const instructions = skill.instructions ?? []
    return [
      `${index + 1}. ${skill.name ?? skill.id ?? 'Unnamed Skill'}`,
      ...instructions.map((instruction: string) => `- ${instruction}`),
    ].join('\n')
  })

  return `${title}:\n${rendered.join('\n\n')}`
}

File: server/models/guidance.py
from typing import Any, Literal

from pydantic import BaseModel, Field

from server.models.seed_skill import SeedSkill


class GuidanceRequest(BaseModel):
    agent_id: str = Field(min_length=1)
    domain: str = Field(min_length=1)
    intent: str = Field(min_length=1)
    context: dict[str, Any] = Field(default_factory=dict)
    risk_level: Literal["low", "medium", "high"] = "low"


class GuidancePayload(BaseModel):
    verified_skills: list[dict[str, Any]] = Field(default_factory=list)
    candidate_skills: list[dict[str, Any]] = Field(default_factory=list)
    seed_skills: list[SeedSkill] = Field(default_factory=list)
    error_patterns: list[str] = Field(default_factory=list)
    output_guidance: list[str] = Field(default_factory=list)
    tool_policy: list[str] = Field(default_factory=list)


class GuidanceResponse(BaseModel):
    id: str
    agent_id: str
    domain: str
    intent: str
    guidance: GuidancePayload
